### PR TITLE
Introduce RHEL based image

### PR DESCRIPTION
**Pull request by riuvshin**
Introduce RHEL based image

### CODE REVIEW

File: dockerfiles/che-fabric8/Dockerfile [hunk 1]
```diff
@@ -4,12 +4,34 @@
 # which accompanies this distribution, and is available at
 # http://www.eclipse.org/legal/epl-v10.html
 
-FROM eclipse/che-server:local
+FROM registry.centos.org/centos/centos:latest
```
[the remark this answers is not in the document]
Comment (riuvshin) [reply dockerfiles/che-fabric8/Dockerfile]: sure, previously we used che image and patch it, but now we will use pure centos image

File: .ci/cico_deploy.sh [hunk 2]
```diff
@@ -68,21 +73,23 @@ docker push "${PROD_IMAGE_DEVSHIFT_LATEST}"
...
-# repository che/che on devshift. The webhook should trigger
-# https://jenkins.cd.test.fabric8.io/che-version-updater/notify 
-# every time a new version of Che is available 
-PROD_IMAGE_DOCKER_HUB="rhche/rh-che-server:${CHE_SERVER_DOCKER_IMAGE_TAG}"
+if [ $TARGET != "rhel" ]; then
+    # We need to continue pushing to the Docker Hub until we have setup a webhook for
+    # repository che/che on devshift. The webhook should trigger
+    # https://jenkins.cd.test.fabric8.io/che-version-updater/notify
+    # every time a new version of Che is available
+    PROD_IMAGE_DOCKER_HUB="rhche/rh-che-server:${CHE_SERVER_DOCKER_IMAGE_TAG}"
```
Comment (ibuziuk) [inline .ci/cico_deploy.sh]: @l0rd @davidfestal I a bit got lost with `https://jenkins.cd.test.fabric8.io/che-version-updater/` do we really need this ?

File: config [hunk 3]
```diff
@@ -1,5 +1,20 @@
 # Config used by the different scripts
 
+TARGET=${TARGET:-"centos"}
+
+if [ $TARGET == "rhel" ]; then
+  export DOCKERFILE="Dockerfile.rhel"
+  export DOCKER_REGISTRY=${DOCKER_REGISTRY:-"prod.registry.devshift.net"}
+  export DOCKER_HUB_NAMESPACE=${DOCKER_REGISTRY}/rhchestage
```
Comment (ibuziuk) [inline config]: @l0rd @davidfestal do we still need `rhchestage` ?

**Comment by davidfestal** [comment]
@riuvshin I don't understand all the changes, and the precise reason for each changes, but I'm worried about the fact that these changes would certainly impact the development scripts that are in `./dev-scripts` (since they use the cico build scripts under-the-cover). Afaict, these changes would also impact the Readme file itself (since it mentions the dev scripts as well as the `DOCKER_HUB_NAMESPACE` environment variable which is removed in your PR).

File: dockerfiles/che-fabric8/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (l0rd) [reply dockerfiles/che-fabric8/Dockerfile]: Ok but what's the link with RHEL based image? Why are you doing this in this PR?

**Comment by riuvshin** [comment]
@davidfestal this changes are basically clean-up with adding logic to handle rhel builds. since I do not see the whole picture I expect on your guys help to get this merged today. I can walk through all changes I did with you if you want and explain what / why I did.

**Pull request merged**